client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx

```diff
@@ -0,0 +1,183 @@
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'calypso/components/data/document-head';
+import EmailHeader from 'calypso/my-sites/email/email-header';
+import {
+	emailManagement,
+	emailManagementTitanControlPanelRedirect,
+} from 'calypso/my-sites/email/paths';
+import EmailPlanHeader from 'calypso/my-sites/email/email-management/home/email-plan-header';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import {
+	getEmailPurchaseByDomain,
+	hasEmailSubscription,
+} from 'calypso/my-sites/email/email-management/home/utils';
+import { getDomainsBySite } from 'calypso/state/sites/domains/selectors';
+import { getSelectedDomain } from 'calypso/lib/domains';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import {
+	hasLoadedSitePurchasesFromServer,
+	isFetchingSitePurchases,
+} from 'calypso/state/purchases/selectors';
+import HeaderCake from 'calypso/components/header-cake';
+import QuerySiteDomains from 'calypso/components/data/query-site-domains';
+import QuerySitePurchases from 'calypso/components/data/query-site-purchases';
+import {
+	TITAN_CONTROL_PANEL_CONTEXT_CONFIGURE_CATCH_ALL_EMAIL,
+	TITAN_CONTROL_PANEL_CONTEXT_CONFIGURE_DESKTOP_APP,
+	TITAN_CONTROL_PANEL_CONTEXT_CONFIGURE_INTERNAL_FORWARDING,
+	TITAN_CONTROL_PANEL_CONTEXT_GET_MOBILE_APP,
+	TITAN_CONTROL_PANEL_CONTEXT_IMPORT_EMAIL_DATA,
+} from 'calypso/lib/titan/constants';
+import VerticalNav from 'calypso/components/vertical-nav';
+import VerticalNavItemEnhanced from 'calypso/components/vertical-nav/item/enhanced';
+
+class TitanManageMailboxes extends Component {
+	static propTypes = {
+		// Props passed to this component
+		selectedDomainName: PropTypes.string.isRequired,
+
+		// Props injected via connect()
+		currentRoute: PropTypes.string.isRequired,
+		domain: PropTypes.object,
+		hasSubscription: PropTypes.bool.isRequired,
+		isLoadingPurchase: PropTypes.bool.isRequired,
+		purchase: PropTypes.object,
+		selectedSite: PropTypes.object.isRequired,
+
+		// Props injected via localize()
+		translate: PropTypes.func.isRequired,
+	};
+
+	handleBack = () => {
+		const { currentRoute, selectedDomainName, selectedSite } = this.props;
+
+		page( emailManagement( selectedSite.slug, selectedDomainName, currentRoute ) );
+	};
+
+	getPath = ( context ) => {
+		const { currentRoute, domain, selectedSite } = this.props;
+
+		if ( ! domain ) {
+			return '';
+		}
+
+		return emailManagementTitanControlPanelRedirect( selectedSite.slug, domain.name, currentRoute, {
+			context,
+		} );
+	};
+
+	render() {
+		const {
+			currentRoute,
+			domain,
+			hasSubscription,
+			isLoadingPurchase,
+			purchase,
+			selectedSite,
+			translate,
+		} = this.props;
+
+		return (
+			<>
+				{ selectedSite && <QuerySiteDomains siteId={ selectedSite.ID } /> }
+
+				{ selectedSite && hasSubscription && <QuerySitePurchases siteId={ selectedSite.ID } /> }
+
+				<DocumentHead title={ translate( 'Manage all mailboxes' ) } />
```

Comment:
Non-blocking: it may be worth adding a `PageViewTracker` component here for the new page, as that doesn't require much work and could be easy for us to forget (as we likely have for the other new pages!)

Reply:
I will add it.